spyder/plugins/ipythonconsole/widgets/namespacebrowser.py

```diff
@@ -50,160 +50,80 @@ class NamepaceBrowserWidget(RichJupyterWidget):
-
-    def configure_namespacebrowser(self):
-        """Configure associated namespace browser widget"""
-        # Update namespace view
-        self.sig_namespace_view.connect(lambda data:
-            self.namespacebrowser.process_remote_view(data))
-
-        # Update properties of variables
-        self.sig_var_properties.connect(lambda data:
-            self.namespacebrowser.set_var_properties(data))
 
     def refresh_namespacebrowser(self):
         """Refresh namespace browser"""
+        if self.kernel_client is None:
+            return
         if self.namespacebrowser:
-            self.silent_exec_method(
-                'get_ipython().kernel.get_namespace_view()')
-            self.silent_exec_method(
-                'get_ipython().kernel.get_var_properties()')
+            self.call_kernel(
+                callback=self.set_namespace_view).get_namespace_view()
+            self.call_kernel(
+                callback=self.set_var_properties).get_var_properties()
+
+    def set_namespace_view(self, view):
+        """Set the current namespace view."""
+        self.namespacebrowser.process_remote_view(view)
+
+    def set_var_properties(self, properties):
+        """Set var properties."""
+        self.namespacebrowser.set_var_properties(properties)
 
     def set_namespace_view_settings(self):
         """Set the namespace view settings"""
+        if self.kernel_client is None:
+            return
         if self.namespacebrowser:
-            settings = to_text_string(
-                self.namespacebrowser.get_view_settings())
-            code =(u"get_ipython().kernel.namespace_view_settings = %s" %
-                   settings)
-            self.silent_execute(code)
+            settings = self.namespacebrowser.get_view_settings()
+            self.call_kernel().set_namespace_view_settings(settings)
 
     def get_value(self, name):
         """Ask kernel for a value"""
-        code = u"get_ipython().kernel.get_value('%s')" % name
-        if self._reading:
-            method = self.kernel_client.input
-            code = u'!' + code
-        else:
-            method = self.silent_execute
-
-        # Wait until the kernel returns the value
-        wait_loop = QEventLoop()
-        self.sig_got_reply.connect(wait_loop.quit)
-        method(code)
-        wait_loop.exec_()
-
-        # Remove loop connection and loop
-        self.sig_got_reply.disconnect(wait_loop.quit)
-        wait_loop = None
-
-        # Handle exceptions
-        if self._kernel_value is None:
-            if self._kernel_reply:
-                msg = self._kernel_reply[:]
-                self._kernel_reply = None
-                raise ValueError(msg)
-
-        return self._kernel_value
+        try:
+            return self.call_kernel(
+                interrupt=True, blocking=True).get_value(name)
+        except (TimeoutError, UnpicklingError):
+            raise ValueError('Kernel did not send the value.')
```

Comment:
This is still problematic because a failure in `get_value` is showing both our error dialog and a QMessageBox that displays the above message. And I only want to show the later here because most of these errors are caused by the inability to pickle the corresponding value in the kernel.

So please make the errors corresponding to `get_value` to not be reported. If you could do that in a generic way (i.e. if we can decide what handlers are allowed to report errors), that'd be even better.

Reply:
After a bit of thought, I think I'm seeing this issue the wrong way by relying in the mental model I have for our previous architecture.

Now I think this is something that needs to be solved in CommBase by addressing failures in pickling the requested data. I'll left you a comment there instead.